# Project 06 -- Anthony Micco

**TA Help:** N/A
    
**Collaboration:** N/A

## Question 1

In [2]:
# loading in the Lahman database

In [3]:
%sql sqlite:////anvil/projects/tdm/data/lahman/lahman.db

In [4]:
# Selecting all the tables in the Lahman database

In [5]:
%%sql
SELECT name FROM sqlite_master WHERE type='table';

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('teamsfranchises',),
 ('teams',),
 ('teamshalf',),
 ('people',),
 ('allstarfull',),
 ('appearances',),
 ('awardsmanagers',),
 ('awardsplayers',),
 ('awardssharemanagers',),
 ('awardsshareplayers',),
 ('batting',),
 ('battingpost',),
 ('schools',),
 ('collegeplaying',),
 ('fielding',),
 ('fieldingof',),
 ('fieldingofsplit',),
 ('fieldingpost',),
 ('halloffame',),
 ('parks',),
 ('homegames',),
 ('managers',),
 ('managershalf',),
 ('pitching',),
 ('pitchingpost',),
 ('salaries',),
 ('seriespost',)]

In [6]:
#Selecting the top 5 records from three different tables

In [7]:
%%sql
SELECT * FROM awardsplayers LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('poseybu01', 'Lou Gehrig Memorial Award', 2019, 'ML', '', ''),
 ('klubeco01', 'Lou Gehrig Memorial Award', 2018, 'ML', '', ''),
 ('vottojo01', 'Lou Gehrig Memorial Award', 2017, 'ML', '', ''),
 ('altuvjo01', 'Lou Gehrig Memorial Award', 2016, 'ML', '', ''),
 ('grandcu01', 'Lou Gehrig Memorial Award', 2015, 'ML', '', '')]

In [8]:
%%sql
SELECT * FROM batting LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('aardsda01', 2004, 1, 'SFN', 'NL', 11, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2006, 1, 'CHN', 'NL', 45, '', 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0),
 ('aardsda01', 2007, 1, 'CHA', 'AL', 25, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2008, 1, 'BOS', 'AL', 47, '', 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0),
 ('aardsda01', 2009, 1, 'SEA', 'AL', 73, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)]

In [9]:
%%sql
SELECT * FROM schools LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('abilchrist', 'Abilene Christian University', 'Abilene', 'TX', 'USA'),
 ('adelphi', 'Adelphi University', 'Garden City', 'NY', 'USA'),
 ('adrianmi', 'Adrian College', 'Adrian', 'MI', 'USA'),
 ('akron', 'University of Akron', 'Akron', 'OH', 'USA'),
 ('alabama', 'University of Alabama', 'Tuscaloosa', 'AL', 'USA')]

1.2) I noticed that the playerID column was in both the awardsplauyers and batting tables, while the rest of the columns were unique to the three tables I selected. \
1.3) The tables might have a variety of different relationships with each other. For example, the people table can have a one to many relationship with one person (playerID) being in many different tables. Additionally, the records in the table could have one to one or many to many relationships where one table relates to exactly one other table or where many tables relate to many other tables. The structure and relationships between tables in a database make up the database schema. 

## Question 2

In [10]:
#Reading in the salaries table

In [11]:
%%sql
SELECT * FROM salaries LIMIT 10

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1985, 'ATL', 'NL', 'barkele01', 870000.0),
 (1985, 'ATL', 'NL', 'bedrost01', 550000.0),
 (1985, 'ATL', 'NL', 'benedbr01', 545000.0),
 (1985, 'ATL', 'NL', 'campri01', 633333.0),
 (1985, 'ATL', 'NL', 'ceronri01', 625000.0),
 (1985, 'ATL', 'NL', 'chambch01', 800000.0),
 (1985, 'ATL', 'NL', 'dedmoje01', 150000.0),
 (1985, 'ATL', 'NL', 'forstte01', 483333.0),
 (1985, 'ATL', 'NL', 'garbege01', 772000.0),
 (1985, 'ATL', 'NL', 'harpete01', 250000.0)]

In [12]:
#Joining the salaries and people tables

In [13]:
%%sql
SELECT * FROM Salaries JOIN People ON Salaries.playerID = People.playerID LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1985, 'ATL', 'NL', 'barkele01', 870000.0, 'barkele01', 1955, 7, 7, 'Fort Knox', 'USA', 'KY', '', '', '', '', '', '', 'Len', 'Barker', 'Leonard Harold', 225, 77, 'R', 'R', '1976-09-14', 'barkele01', '1987-09-26', 'barkl001'),
 (1985, 'ATL', 'NL', 'bedrost01', 550000.0, 'bedrost01', 1957, 12, 6, 'Methuen', 'USA', 'MA', '', '', '', '', '', '', 'Steve', 'Bedrosian', 'Stephen Wayne', 200, 75, 'R', 'R', '1981-08-14', 'bedrost01', '1995-08-09', 'bedrs001'),
 (1985, 'ATL', 'NL', 'benedbr01', 545000.0, 'benedbr01', 1955, 8, 18, 'Birmingham', 'USA', 'AL', '', '', '', '', '', '', 'Bruce', 'Benedict', 'Bruce Edwin', 175, 73, 'R', 'R', '1978-08-18', 'benedbr01', '1989-09-11', 'beneb001'),
 (1985, 'ATL', 'NL', 'campri01', 633333.0, 'campri01', 1953, 6, 10, 'Trion', 'USA', 'GA', 2013, 4, 25, 'USA', 'GA', 'Rydal', 'Rick', 'Camp', 'Rick Lamar', 195, 73, 'R', 'R', '1976-09-15', 'campri01', '1985-10-05', 'campr001'),
 (1985, 'ATL', 'NL', 'ceronri01', 625000.0, 'ceronri01', 1954, 5, 19, 'Newark', 'USA', 'NJ', '', '', '', '', '', '', 'Rick', 'Cerone', 'Richard Aldo', 192, 71, 'R', 'R', '1975-08-17', 'ceronri01', '1992-07-10', 'ceror001')]

In [14]:
#Joining the salaries and people tables as a temporary table

In [15]:
%%sql
WITH joined_data AS (
    SELECT * FROM Salaries JOIN People ON People.playerID = Salaries.playerID
)

SELECT * FROM joined_data LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1985, 'ATL', 'NL', 'barkele01', 870000.0, 'barkele01', 1955, 7, 7, 'Fort Knox', 'USA', 'KY', '', '', '', '', '', '', 'Len', 'Barker', 'Leonard Harold', 225, 77, 'R', 'R', '1976-09-14', 'barkele01', '1987-09-26', 'barkl001'),
 (1985, 'ATL', 'NL', 'bedrost01', 550000.0, 'bedrost01', 1957, 12, 6, 'Methuen', 'USA', 'MA', '', '', '', '', '', '', 'Steve', 'Bedrosian', 'Stephen Wayne', 200, 75, 'R', 'R', '1981-08-14', 'bedrost01', '1995-08-09', 'bedrs001'),
 (1985, 'ATL', 'NL', 'benedbr01', 545000.0, 'benedbr01', 1955, 8, 18, 'Birmingham', 'USA', 'AL', '', '', '', '', '', '', 'Bruce', 'Benedict', 'Bruce Edwin', 175, 73, 'R', 'R', '1978-08-18', 'benedbr01', '1989-09-11', 'beneb001'),
 (1985, 'ATL', 'NL', 'campri01', 633333.0, 'campri01', 1953, 6, 10, 'Trion', 'USA', 'GA', 2013, 4, 25, 'USA', 'GA', 'Rydal', 'Rick', 'Camp', 'Rick Lamar', 195, 73, 'R', 'R', '1976-09-15', 'campri01', '1985-10-05', 'campr001'),
 (1985, 'ATL', 'NL', 'ceronri01', 625000.0, 'ceronri01', 1954, 5, 19, 'Newark', 'USA', 'NJ', '', '', '', '', '', '', 'Rick', 'Cerone', 'Richard Aldo', 192, 71, 'R', 'R', '1975-08-17', 'ceronri01', '1992-07-10', 'ceror001')]

In [16]:
#Joining salaries and people tables, finding the average salary of peoples grouped by weight and sorted in descending order

In [17]:
%%sql
WITH joined_data AS (
    SELECT * FROM Salaries JOIN People ON People.playerID = Salaries.playerID
)

SELECT *, AVG(salary) AS avg_salary FROM joined_data GROUP BY weight ORDER BY weight DESC LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(2015, 'CIN', 'NL', 'diazju03', 510000.0, 'diazju03', 1984, 2, 27, 'La Romana', 'D.R.', 'La Romana', '', '', '', '', '', '', 'Jumbo', 'Diaz', 'Jose Rafael', 315, 76, 'R', 'R', '2014-06-20', 'diazju03', '2017-07-16', 'diazj005', 517500.0),
 (2002, 'CLE', 'AL', 'sabatcc01', 700000.0, 'sabatcc01', 1980, 7, 21, 'Vallejo', 'USA', 'CA', '', '', '', '', '', '', 'CC', 'Sabathia', 'Carsten Charles', 300, 78, 'L', 'L', '2001-04-08', 'sabatc.01', '2019-09-24', 'sabac001', 14576190.4),
 (1996, 'SLN', 'NL', 'youngdm01', 109000.0, 'youngdm01', 1973, 10, 11, 'Vicksburg', 'USA', 'MS', '', '', '', '', '', '', 'Dmitri', 'Young', 'Dmitri Dell', 295, 74, 'B', 'R', '1996-08-29', 'youngdm01', '2008-07-11', 'yound001', 3771714.285714286),
 (2002, 'CHA', 'AL', 'rauchjo01', 200000.0, 'rauchjo01', 1978, 9, 27, 'Louisville', 'USA', 'KY', '', '', '', '', '', '', 'Jon', 'Rauch', 'Jon Erich', 290, 83, 'R', 'R', '2002-04-02', 'rauchjo01', '2013-05-17', 'raucj001', 1613250.0),
 (1997, 'CLE', 'AL', 'colonba01', 150000.0, 'colonba01', 1973, 5, 24, 'Altamira', 'D.R.', 'Puerto Plata', '', '', '', '', '', '', 'Bartolo', 'Colon', 'Bartolo', 285, 71, 'R', 'R', '1997-04-04', 'colonba01', '2018-09-22', 'colob001', 5617744.680851064),
 (2002, 'BOS', 'AL', 'pickeca01', 200000.0, 'pickeca01', 1976, 9, 29, 'St. Thomas', 'V.I.', 'St. Thomas', '', '', '', '', '', '', 'Calvin', 'Pickering', 'Calvin Elroy', 283, 77, 'L', 'L', '1998-09-12', 'pickeca01', '2005-04-21', 'pickc001', 261750.0),
 (2002, 'PHI', 'NL', 'silvaca01', 200000.0, 'silvaca01', 1979, 4, 23, 'Bolivar', 'Venezuela', 'Bolivar', '', '', '', '', '', '', 'Carlos', 'Silva', 'Carlos', 280, 76, 'R', 'R', '2002-04-01', 'silvaca01', '2010-09-07', 'silvc001', 2301871.0),
 (2003, 'CHN', 'NL', 'zambrca01', 340000.0, 'zambrca01', 1981, 6, 1, 'Puerto Cabello', 'Venezuela', 'Carabobo', '', '', '', '', '', '', 'Carlos', 'Zambrano', 'Carlos Alberto', 275, 76, 'B', 'R', '2001-08-20', 'zambrca01', '2012-09-21', 'zambc001', 10164293.103448275),
 (2016, 'MIN', 'AL', 'sanomi01', 547500.0, 'sanomi01', 1993, 5, 11, 'San Pedro de Macoris', 'D.R.', 'San Pedro de Macoris', '', '', '', '', '', '', 'Miguel', 'Sano', 'Miguel Angel', 272, 76, 'R', 'R', '2015-07-02', 'sanomi01', '2022-07-29', 'sanom001', 547500.0),
 (1999, 'CHA', 'AL', 'leeca01', 200000.0, 'leeca01', 1976, 6, 20, 'Aguadulce', 'Panama', 'Cocle', '', '', '', '', '', '', 'Carlos', 'Lee', 'Carlos', 270, 74, 'R', 'R', '1999-05-07', 'leeca01', '2012-10-03', 'lee-c001', 4128199.1228070175)]

## Question 3

In [18]:
# reading in the batting table

In [19]:
%%sql
SELECT * FROM batting LIMIT 10

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('aardsda01', 2004, 1, 'SFN', 'NL', 11, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2006, 1, 'CHN', 'NL', 45, '', 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0),
 ('aardsda01', 2007, 1, 'CHA', 'AL', 25, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2008, 1, 'BOS', 'AL', 47, '', 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0),
 ('aardsda01', 2009, 1, 'SEA', 'AL', 73, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2010, 1, 'SEA', 'AL', 53, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2012, 1, 'NYA', 'AL', 1, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2013, 1, 'NYN', 'NL', 43, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2015, 1, 'ATL', 'NL', 33, '', 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0),
 ('aaronha01', 1954, 1, 'ML1', 'NL', 122, '', 468, 58, 131, 27, 6, 13, 69, 2, 2, 28, 39, '', 3, 6, 4, 13)]

In [30]:
#joining the batting and people tables and finding how many runs Babe Ruth scored during his career

In [31]:
%%sql
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)

SELECT SUM(R) FROM joined_data WHERE nameFirst = 'Babe' AND nameLast = 'Ruth'

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(2174,)]

In [ ]:
# writing a query to find the average number of runs scored by players in the 1980s

In [41]:
%%sql
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)

SELECT birthYear, AVG(R) AS avg_runs FROM joined_data WHERE nameLast NOT LIKE '%�%' AND birthYear >= 1980 AND birthYear < 1990 GROUP BY birthYear ORDER BY birthYear DESC

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1989, 12.811985018726592),
 (1988, 9.934713375796179),
 (1987, 14.006301197227474),
 (1986, 13.602470930232558),
 (1985, 12.153616703952274),
 (1984, 13.758029978586723),
 (1983, 16.491117478510027),
 (1982, 16.083526682134572),
 (1981, 14.335890878090366),
 (1980, 17.10182767624021)]

## Question 4

In [42]:
# query to find the players with the most runs scored in their careers

In [54]:
%%sql

WITH joined_data AS (
    SELECT * FROM people JOIN batting ON people.playerID = batting.playerID
)

SELECT playerID,nameFirst,nameLast, SUM(R) AS totalRuns FROM joined_data WHERE nameFirst NOT LIKE '%�%' AND nameLast NOT LIKE '%�%' GROUP BY playerID ORDER BY totalRuns DESC LIMIT 10

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('henderi01', 'Rickey', 'Henderson', 2295),
 ('cobbty01', 'Ty', 'Cobb', 2247),
 ('bondsba01', 'Barry', 'Bonds', 2227),
 ('ruthba01', 'Babe', 'Ruth', 2174),
 ('aaronha01', 'Hank', 'Aaron', 2174),
 ('rosepe01', 'Pete', 'Rose', 2165),
 ('mayswi01', 'Willie', 'Mays', 2062),
 ('rodrial01', 'Alex', 'Rodriguez', 2021),
 ('ansonca01', 'Cap', 'Anson', 1999),
 ('musiast01', 'Stan', 'Musial', 1949)]

In [57]:
# query to find the players with the most hits in their careers

In [58]:
%%sql

WITH joined_data AS (
    SELECT * FROM people JOIN batting ON people.playerID = batting.playerID
)

SELECT playerID,nameFirst,nameLast, SUM(H) AS totalHits FROM joined_data WHERE nameFirst NOT LIKE '%�%' AND nameLast NOT LIKE '%�%' GROUP BY playerID ORDER BY totalHits DESC LIMIT 10

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('rosepe01', 'Pete', 'Rose', 4256),
 ('cobbty01', 'Ty', 'Cobb', 4189),
 ('aaronha01', 'Hank', 'Aaron', 3771),
 ('musiast01', 'Stan', 'Musial', 3630),
 ('speaktr01', 'Tris', 'Speaker', 3514),
 ('jeterde01', 'Derek', 'Jeter', 3465),
 ('ansonca01', 'Cap', 'Anson', 3435),
 ('wagneho01', 'Honus', 'Wagner', 3420),
 ('yastrca01', 'Carl', 'Yastrzemski', 3419),
 ('pujolal01', 'Albert', 'Pujols', 3384)]

4.3) There are 5 players who appear in both lists: Pete Rose, Ty Cobb, Hank Aaron, Stan Musial, and Cap Anson

## Question 5

In [59]:
# reading in the CollegePlaying table 

In [60]:
%%sql
SELECT * FROM CollegePlaying LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('aardsda01', 'pennst', 2001),
 ('aardsda01', 'rice', 2002),
 ('aardsda01', 'rice', 2003),
 ('abadan01', 'gamiddl', 1992),
 ('abadan01', 'gamiddl', 1993)]

In [61]:
#joining the CollegePlaying and batting tables to find the most runs scored by Purdue University players 

In [68]:
%%sql
WITH joined_data AS (
    SELECT CollegePlaying.playerID, Batting.R FROM CollegePlaying JOIN Batting ON CollegePlaying.playerID = Batting.playerID  WHERE CollegePlaying.schoolID = 'purdue'
)

SELECT *, SUM(R) AS careerRuns FROM joined_data GROUP BY playerID ORDER BY careerRuns DESC LIMIT 10

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('allenbe01', 2, 1071),
 ('skowrbi01', 0, 681),
 ('allenje01', 9, 465),
 ('mackife01', 0, 255),
 ('rossyri01', 0, 172),
 ('cianfar01', 3, 136),
 ('tragewa01', 0, 108),
 ('mccabjo01', 1, 6),
 ('kellysp01', 3, 3),
 ('wisehu01', 0, 0)]

In [69]:
#query to find the average number of runs scored by players from a university

In [73]:
%%sql
WITH joined_data AS (
    SELECT * FROM CollegePlaying JOIN Batting ON CollegePlaying.playerID = Batting.playerID
)

SELECT schoolID, AVG(R) AS avgRuns FROM joined_data GROUP BY schoolID ORDER BY avgRuns DESC LIMIT 10

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('hartford', 86.24528301886792),
 ('txwesleyan', 78.45833333333333),
 ('momaple', 75.1923076923077),
 ('ecentralok', 73.95454545454545),
 ('montevallo', 71.44444444444444),
 ('narizona', 70.15384615384616),
 ('wittnbrgoh', 68.0909090909091),
 ('illchicago', 67.61111111111111),
 ('inconco', 66.32352941176471),
 ('nmmilin', 65.0)]

5.2) The highest number of runs scored by a player from Purdue University is 1071 \
5.4) The university with the highest average runs is Hartford University with approx. 86.25 runs 

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.